ForcedMount update

thebrodieman opened the pull request:
Chosen flying mount will now match that of user defined settings, if there is one chosen.

thebrodieman committed:
ForcedMount update

Chosen flying mount will now match that of user defined settings, if
there is one chosen.

jakobbotsch commented on Quest Behaviors/ForcedMount.cs:
```diff
@@ -110,12 +110,19 @@ public ForcedMount(Dictionary<string, string> args)
             // Hop off the ground. Kthx
-            await Coroutine.Sleep(2500);
+            await Coroutine.Sleep(250);
+            await Coroutine.Wait(2000, () => StyxWoW.Me.Mounted);
```
There is already a wait for this on line 120. Removing this line but keeping the 250 ms sleep instead of 2500 ms would be better.

daydos replied:
Those can completely be removed when Flightor.MountHelper.MountUp is used.

jakobbotsch commented on Quest Behaviors/ForcedMount.cs:
```diff
@@ -110,12 +110,19 @@ public ForcedMount(Dictionary<string, string> args)
+                        Mount.FlyingMounts.FirstOrDefault(m => m.CreatureSpellId == mount.CreatureSpellId)?.CreatureSpell.Cast();
+                }
+                else
+                    Mount.FlyingMounts.First().CreatureSpell.Cast();
```
All of this can use [Flightor.MountHelper.MountUp](http://docs.honorbuddy.com/html/9a7d751f-71d6-a362-aba6-d6ee192432be.htm) instead, I believe.

daydos replied:
Indeed, Flightor.MountHelper.MountUp can be used here. It handles shapeshifts, preferred mounts etc. already.

thebrodieman committed:
Usage of Flightor's Mountup added

Exchanged out older code along with new update to utilize Flightor's
MountUp() method. Tested and uses user definition mount correctly.

jakobbotsch commented on Quest Behaviors/ForcedMount.cs:
```diff
@@ -99,23 +99,11 @@ public ForcedMount(Dictionary<string, string> args)
-                await Coroutine.Wait(3000, () => StyxWoW.Me.Mounted);
-            }
+            Flightor.MountHelper.MountUp();
+            await Coroutine.Wait(3000, () => StyxWoW.Me.Mounted);
```
MountUp already waits for the mount, so just get rid of this wait. We will clean all of this up a bit
in the near future.

thebrodieman committed:
Removal of unnecessary Coroutine.Wait

Removed 3 second wait timer awaiting if we are mounted. MountUp()
already innnately does this.

jakobbotsch commented:
Thanks!
